src/dsmr/fields.h

```diff
@@ -167,6 +167,18 @@ const uint8_t WATER_MBUS_ID = 2;
 const uint8_t THERMAL_MBUS_ID = 3;
 const uint8_t SLAVE_MBUS_ID = 4;
 
+#ifdef ARDUINO_ARCH_ESP8266
+#define DEFINE_FIELD(fieldname, value_t, obis, field_t, field_args...) \
+  struct fieldname : field_t<fieldname, ##field_args> { \
+    value_t fieldname; \
+    bool fieldname ## _present = false; \
+    static constexpr ObisId id = obis; \
+    static constexpr char name_progmem[] = #fieldname; \
+    static constexpr const __FlashStringHelper *name = reinterpret_cast<const __FlashStringHelper*>(&name_progmem); \
+    value_t& val() { return fieldname; } \
+    bool& present() { return fieldname ## _present; } \
+  }
+#else
```

Comment:
It seems this is identical, except for the missing PROGMEM. I think it would be better to define a `DSMR_PROGMEM` macro (set it to `PROGMEM` when it is defind and not on ESP, define it empty otherwise) and then use `DSMR_PROGMEM` everywhere instead of just `PROGMEM`. Be sure to include a comment, ideally pointing to a bug report for the bug this works around.

Reply:
I'm in doubt about putting this in util.h or in a separate file. What do you think?

Reply:
`util.h` seems fine to me.